ml_peg/analysis/supramolecular/LNCI16/analyse_LNCI16.py

```diff
@@ -0,0 +1,248 @@
+            if xyz_files:
+                is_charged = []
+                for xyz_file in xyz_files:
+                    atoms = read(xyz_file)
+                    charge = atoms.info.get("complex_charge", 0)
+                    is_charged.append(charge != 0)
+                return is_charged
+    return []
+
+
+@pytest.fixture
+@plot_parity(
+    filename=OUT_PATH / "figure_interaction_energies.json",
+    title="LNCI16 Interaction Energies",
+    x_label="Predicted interaction energy / kcal/mol",
+    y_label="Reference interaction energy / kcal/mol",
+    hoverdata={
+        "System": get_system_names(),
+        "Complex Atoms": get_atom_counts(),
+        "Charge": get_charges(),
+        "Charged": get_is_charged(),
+    },
+)
+def interaction_energies() -> dict[str, list]:
+    """
+    Get interaction energies for all LNCI16 systems.
+
+    Returns
+    -------
+    dict[str, list]
+        Dictionary of reference and predicted interaction energies.
+    """
+    from ase.io import read
+
+    results = {"ref": []} | {mlip: [] for mlip in MODELS}
+    ref_stored = False
+
+    for model_name in MODELS:
+        model_dir = CALC_PATH / model_name
+
+        if not model_dir.exists():
+            results[model_name] = []
+            continue
+
+        xyz_files = sorted(model_dir.glob("*.xyz"))
+        if not xyz_files:
+            results[model_name] = []
+            continue
+
+        model_energies = []
+        ref_energies = []
+
+        for xyz_file in xyz_files:
+            atoms = read(xyz_file)
+            model_energies.append(atoms.info["E_int_model_kcal"])
+            if not ref_stored:
+                ref_energies.append(atoms.info["E_int_ref_kcal"])
+
+        results[model_name] = model_energies
+
+        # Store reference energies (only once)
+        if not ref_stored:
+            results["ref"] = ref_energies
+            ref_stored = True
+
+        # Copy individual structure files to app data directory
+        structs_dir = OUT_PATH / model_name
+        structs_dir.mkdir(parents=True, exist_ok=True)
+
+        # Copy individual structure files
+        import shutil
+
+        for i, xyz_file in enumerate(xyz_files):
+            shutil.copy(xyz_file, structs_dir / f"{i}.xyz")
+
+    return results
+
+
+@pytest.fixture
+def lnci16_mae(interaction_energies) -> dict[str, float]:
+    """
+    Get mean absolute error for interaction energies.
+
+    Parameters
+    ----------
+    interaction_energies
+        Dictionary of reference and predicted interaction energies.
+
+    Returns
+    -------
+    dict[str, float]
+        Dictionary of predicted interaction energy errors for all models.
+    """
+    results = {}
+    for model_name in MODELS:
+        if interaction_energies[model_name]:
+            results[model_name] = mae(
+                interaction_energies["ref"], interaction_energies[model_name]
+            )
+        else:
+            results[model_name] = float("nan")
```

Comment:
Do we do this anywhere else?

In general I've been starting to use `None` for missing values, to distinguish more from potential numerical errors that may lead to NaNs.